### PR TITLE
fix axis arguments for concat

### PR DESCRIPTION
**Pull request by Dr-Irv**
pyright 2021.5.4 has issue with `pd.concat` return types based on axes:
```python
import pandas as pd

s1 = pd.Series([1, 2, 3])
s2 = pd.Series([4, 5, 6])

df: pd.DataFrame = pd.concat([s1, s2], axis=1)
```
reports
```
Expression of type "Series[Dtype@concat]" cannot be assigned to declared type "DataFrame"
  "Series[Dtype@concat]" is incompatible with "DataFrame"
```

This PR fixes that issue.  Basically the `axis` argument determines return types when doing `pd.concat()` on `Series`

### CODE REVIEW

File: pandas/core/reshape/concat.pyi [hunk 1]
```diff
@@ -4,9 +4,11 @@ from typing import Hashable, Iterable, Literal, Mapping, Optional, Union, overlo
 
 
 @overload
-def concat(objs: Union[Iterable[Optional[Series]], Mapping[Optional[Hashable], Optional[Series]]], join: str=..., ignore_index: bool=..., keys=..., levels=..., names=..., verify_integrity: bool=..., sort: bool=..., copy: bool=..., *, axis: Literal[0]) -> Series[Dtype]: ...
+def concat(objs: Union[Iterable[Optional[Series]], Mapping[Optional[Hashable], Optional[Series]]], join: str=..., ignore_index: bool=..., keys=..., levels=..., names=..., verify_integrity: bool=..., sort: bool=..., copy: bool=..., axis::Literal[0,'index',1,'columns']=Union[0,'index']) -> Series[Dtype]: ...
```
Comment (jakebailey) [inline pandas/core/reshape/concat.pyi:7]: Typo: double `:` at axis.

**Comment by Dr-Irv** [comment]
Turns out that I didn't fix it, and I can't figure out how.  So I am going to create an issue.  #69

**Comment by Dr-Irv** [comment]
@jakebailey I reopened this as I figured out the error I had, which was related to how to import `Literal`

**Comment by jakebailey** [comment]
Hm, `Literal` should have been equivalent imported in the original way; in pyright we largely assume that `typing` has `Literal` already. Are you sure the fix wasn't some other change that was made?

In any case, I'll test this out and merge before next week's Pylance release.

**Comment by Dr-Irv** [comment]
> Hm, `Literal` should have been equivalent imported in the original way; in pyright we largely assume that `typing` has `Literal` already. Are you sure the fix wasn't some other change that was made?

Yes.  I was getting the `pyright` errors if I did this:
```python
from pandas._typing import Dtype
from typing import Hashable, Iterable, Literal, Mapping, Optional, Union, overload
```
When I changed it to:
```python
from pandas._typing import Dtype, Literal
from typing import Hashable, Iterable, Mapping, Optional, Union, overload
```
then everything worked.  Just did a test and verified that, so you could try as well.

The conda environment (which I have to use for my project) is using python 3.7, so that might explain it.  If the pyi file imports `Literal` in a 3.7 environment, I think that is messing up matching arguments that are `Literal`.

**Review by jakebailey — APPROVED** [review]

**Comment by jakebailey** [comment]
Sorry for looking at this so late. This appears to not work:

image

Swapping the lines doesn't either, it seems.

**Comment by Dr-Irv** [comment]
> Sorry for looking at this so late. This appears to not work:

Just pushed a commit that addresses that.  Note that I reformatted the PYI file to make it easier for me to see what was going on.

Wasn't sure how to make it so that `axis=1` and `axis="columns"` would do the same thing without stating the overload twice, so I took that route.

**Comment by jakebailey** [comment]
Thanks.

**Review by jakebailey — APPROVED** [review]

**Pull request merged**